core/store/src/main/java/uk/gov/gchq/gaffer/store/SchemaOperationChainValidator.java

```diff
@@ -47,7 +47,7 @@ protected Schema getSchema(final Operation operation, final User user, final Sto
     }
 
     @Override
-    protected Set<StoreTrait> getStoreTraits(final Store store) {
+    protected Set<StoreTrait> getStoreTraits(final Store store, final Context context) {
```

Comment:
From what I can tell it could be because when validating a schema all traits should be considered. The way this is implemented is not all that clear. I think it would be better to remove this class and refactor, this probably wouldn't be much more work than documenting this. Something for a new issue.